Add authentication docs from the customer perspective (#1931)
NuGet/docs.microsoft.com-nuget: docs/reference/extensibility/NuGet-Credential-Providers-for-Visual-Studio.md

A sample implementation can be found in [the VsCredentialProvider sample](https://github.com/NuGet/Samples/tree/master/VsCredentialProvider).

> The NuGet Visual Studio Extension uses an internal `VsCredentialProviderImporter` which also scans for plug-in credential providers. These plug-in credential providers must be discoverable as a MEF Export of type `IVsCredentialProvider`.

Reviewer: As mentioned in an earlier comment, I think calling NuGet an extension might be confusing.

```suggestion
Within Visual Studio, NuGet uses an internal `VsCredentialProviderImporter` which also scans for plug-in credential providers. These plug-in credential providers must be discoverable as a MEF Export of type `IVsCredentialProvider`.
```
